ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Validations.RetrieveById.cs

```diff
@@ -0,0 +1,49 @@
+﻿// ------------------------------------------------
+// Copyright (c) MumsWhoCode. All rights reserved.
+// ------------------------------------------------
+
+using System;
+using Moq;
+using ShoppingList.ConsoleApp.Models.ShoppingItems;
+using ShoppingList.ConsoleApp.Models.ShoppingItems.Exceptions;
+using Xunit;
+
+namespace ShoppingList.Tests.Unit.Services.Foundations.ShoppingItems
+{
+    public partial class ShoppingItemServiceTests
+    {
+        [Fact]
+        public void ShouldThrowValidationExceptionOnRetrieveByIdIfIdIsInvalidAndLogIt()
+        {
+            // given
+            int invalidId = default;
+            var invalidShoppingItemException = new InvalidShoppingItemException();
+
+            invalidShoppingItemException.AddData(
+                key: nameof(ShoppingItem.Id),
+                values: "Value is required.");
```

Comment:
Id is required ?

Reply:
Value is required is a common message for id and quantity. Thanks.